### PR TITLE
Simplify fetching logic

### PR DESCRIPTION
**Pull request by roncohen**
- Remove retry-logic that was too complicated. We will no longer cache negative responses at all.
- Increase expire time from 7 to 30 days.
- Don't block on LiveSatisfaction initialization

### CODE REVIEW

File: packages/browser-sdk/src/feature/features.ts [hunk 1]
```diff
@@ -177,40 +178,60 @@ export class FeaturesClient {
...
+          .catch(() => {
+            // we don't care about the result, we just want to re-fetch
+          });
         return cachedItem.features;
       }
+    }
+
+    // if there's no cached item or there is a stale one but `staleWhileRevalidate` is disabled
+    // try fetching a new one
+    const fetchedFlags = await this.fetchFeatures();
```
Comment (pavkam) [inline packages/browser-sdk/src/feature/features.ts:208]: I see a mixture of "flags" and "features" in the names. Probably better to switch to features in every name and comment for clarity

**Review by pavkam — APPROVED** [review]
Just to confirm: we have opted out from having an immediate retry mechanism. Instead we will poll for flags in background but also try to get fresh ones anytime when we can without having a "max tries"

File: packages/browser-sdk/src/feature/features.ts (hunk 1 above)
Comment (roncohen) [reply packages/browser-sdk/src/feature/features.ts:208]: good catch. Missed that one 👍

**Comment by roncohen** [comment]
> Just to confirm: we have opted out from having an immediate retry mechanism. Instead we will poll for flags in background but also try to get fresh ones anytime when we can without having a "max tries"

For feature/flag fetching, there was never an immediate retry actually. It would only try on page loads. If there was multiple failed loads within 60s it would eventually give up and cache a "negative" result and would then use that for 60s. This was super confusing and the code was error prone. There's also no background refresh yet. I figured we'd go straight to #165 instead.

**Pull request merged**